File: Easy/3396. Minimum Number of Operations to Make Elements in Array Distinct/Java/BruteForce.java
// Time Complexity: O(N²)
// Space Complexity: O(N)

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

class BruteForce {
    public int minimumOperations(int[] nums) {
        short ans = 0;
        List<Integer> list = new ArrayList<>();
        for (int num : nums) {
            list.add(num);  // Add each element of the array to the list
        }
        Set<Integer> set = new HashSet<>(list);

        while(list.size() != set.size()){
            ans+=1;
            if(list.size() <= 3){
                return ans;
            }
            list = list.subList(3, list.size());
            set = new HashSet<>(list);
        }

        return ans;
    }

     public static void main(String[] args) {
        BruteForce solution = new BruteForce();

        // Test case 1
        int[] nums1 = {1, 2, 3, 4, 2, 3, 3, 5, 7};
        System.out.println("Test Case 1: " + solution.minimumOperations(nums1));  // Expected output: 2

        // Test case 2
        int[] nums2 = {4, 5, 6, 4, 4};
        System.out.println("Test Case 2: " + solution.minimumOperations(nums2));  // Expected output: 1

        // Test case 3
        int[] nums3 = {5, 4, 3, 2, 1};
        System.out.println("Test Case 3: " + solution.minimumOperations(nums3));  // Expected output: 0

        // Test case 4
        int[] nums4 = {1, 2, 3, 4, 5, 6, 7, 8, 9};
        System.out.println("Test Case 4: " + solution.minimumOperations(nums4));  // Expected output: 0

        // Test case 5
        int[] nums5 = {6, 7, 8, 9};
        System.out.println("Test Case 5: " + solution.minimumOperations(nums5));  // Expected output: 0
    }
}

File: Meduim/2799. Count Complete Subarrays in an array/Python/bruteForce.py
def countCompleteSubarrays(nums) -> int:
    count = 0
    n = len(nums)
    hashset = set(nums)
    k = len(hashset)

    if k == 1:
        count = (n * (n+1)) // 2
        return count
    for j in range(k, j+1):
        for i in range(n - j + 1):
            if set(nums[i:i+j]) == hashset:
                count += 1

    return count 


if __name__ == "__main__":
    nums1 = [1, 3, 1, 2, 2]
    nums2 = [5, 5, 5, 5]

    print(f'The result of the 1st test case : {countCompleteSubarrays(nums1)}')
    print(f'The result of the 2nd test case : {countCompleteSubarrays(nums2)}')